### DISCUSSION TITLE
List of things to discuss that came up while writing the new tutorials

### DISCUSSION THREAD

**Discussion by vhirtham** [BAMWelDX/weldx#559] [General]
Before I open an issue for every single thing I notice during the creation of the new tutorials, I think it is easier to gather and discuss them here first. We can then create issues if needed.

**Welding current and voltage in single-pass weld schema**

Welding current and voltage are currently stored at the top level of the single-pass weld schema for faster access. They are stored as `Signal`, which I think does not make sense outside of the `MeasurementChain`. My suggestion would be to store it as `TimeSeries`. This would only require us to add one `.data` during the serialization. 

**Add `Geometry.from_profile`**

Most of our examples and actual use cases just have linear traces. I think it would be reasonable to add a special `from_profile` or `from_profile_and_length` method to the `Geometry` so that the user can get a quick 3d visualization without the need to mess around with traces and trace segments.

**Comment by CagtayFabry** [comment]
> Before I open an issue for every single thing I notice during the creation of the new tutorials, I think it is easier to gather and discuss them here first. We can then create issues if needed.
> 
> **Welding current and voltage in single-pass weld schema**
> 
> Welding current and voltage are currently stored at the top level of the single-pass weld schema for faster access. They are stored as `Signal`, which I think does not make sense outside of the `MeasurementChain`. My suggestion would be to store it as `TimeSeries`. This would only require us to add one `.data` during the serialization.

~~I think we could also swap the `Signal` class and simply inherit from `TimeSeries` with the `signal_type` stored in `wx_metadata`.
We don't really use the `Signal` class on its own anyway~~

update: ok, disregard my comment about the usage of the Signal class, we need that for defining 'empty' Signals without TimeSeries data attached.
certainly would make sense to link a `TimeSeries` because that would also ensure that we have an actual recording

**Reply by marscher** [reply]
The duality of discussions and issues... I agree upon attaching/linking the "plain" Timeseries object.

**Comment by vhirtham** [comment]
> Most of our examples and actual use cases just have linear traces. I think it would be reasonable to add a special `from_profile` or `from_profile_and_length` method to the `Geometry` so that the user can get a quick 3d visualization without the need to mess around with traces and trace segments.

Open for discussion - but I think I will just create PR cause this can be implemented rather quick and I don't think anyone will have a problem with it 😉

**Reply by CagtayFabry** [reply]
yes sounds good 👍 
so this function would also create the simplest 3d rasterization? (only start and end points)

**Reply by vhirtham** [reply]
That's the plan
